src/pages/VaultsPage/VaultRow.tsx

```diff
@@ -45,6 +46,15 @@ export function VaultRow({
           onDeposit={updateBalance}
         />
       ) : null}
+      {modal === 'withdraw' ? (
+        <WithdrawModal
+          visible={true}
+          onClose={closeModal}
+          vault={vault}
+          balance={balance}
+          onWithdraw={updateBalance}
+        />
+      ) : null}
```

Comment:
should we change the modal render to switch/case?

Reply:
Good idea